src/client/components/HealthCareApp.jsx

```diff
@@ -65,6 +74,26 @@ class HealthCareApp extends React.Component {
     });
   }
 
+  redirect() {
+    const previousPaths = [];
+    const currentPath = this.props.location.pathname;
+    const paths = _.keys(this.props.uiState.sections);
+
+    for (let i = 0; i < paths.length; i++) {
```

Comment:
I think this is something that could be written a bit more effectively with a map but if this is working I approve, don't want to hold this up by prematurely optimizing.